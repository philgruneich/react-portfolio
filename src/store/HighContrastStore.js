import React, { useState, createContext } from 'react';

export const HighContrastContext = createContext();
HighContrastContext.displayName = 'HighContrastContext';

export const HighContrastStore = props => {
  const [hasContrast, saveHasContrast] = useState(false);

  const toggleContrast = () => {
    saveHasContrast(() => !hasContrast);
  }

  return (
    <HighContrastContext.Provider
      value={{
        state: hasContrast,
        dispatch: saveHasContrast,
        toggle: toggleContrast
      }}
      {...props}
    />
  )
}
