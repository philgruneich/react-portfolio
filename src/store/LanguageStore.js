import React, { useState, useEffect, createContext } from 'react';
import { languages } from '~/lang';

export const LanguageContext = createContext();
LanguageContext.displayName = 'LanguageContext';

export const LanguageStore = props => {
  const [lang, saveLang] = useState('en');
  const [translations, saveTranslations] = useState(languages[lang]);

  useEffect(() => {
    saveTranslations(languages[lang]);
  }, [lang]);

  return (
    <LanguageContext.Provider
      value={{
        state: { lang, translations },
        dispatch: saveLang
      }}
      {...props}
    />
  )
}
