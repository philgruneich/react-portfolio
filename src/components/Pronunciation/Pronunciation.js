import React, { useRef, useContext } from 'react';
import { HighContrastContext } from '~/store/HighContrastStore';
import {
  PronunciationButton,
  PronunciationSvg,
  PronunciationPath
} from './Pronunciation.styles';

export const Pronunciation = ({ src }) => {
  const contrastContext = useContext(HighContrastContext);
  const audioRef = useRef();

  const onClickHandler = event => {
    event.preventDefault();

    audioRef.current.play();
  };

  return (
    <PronunciationButton onClick={onClickHandler} aria-label="Play sound">
      <audio ref={audioRef} src={src} crossOrigin="anonymous" type="audio/mp3" />
      <PronunciationSvg>
        <PronunciationPath hasHighContrast={contrastContext.state} />
      </PronunciationSvg>
    </PronunciationButton>
  )
}
