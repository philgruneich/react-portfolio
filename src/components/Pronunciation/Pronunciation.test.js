import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { HighContrastStore } from '~/store/HighContrastStore';
import { Pronunciation } from './Pronunciation';

const defaultHighContrastState = {
  state: false
}

test('Matches snapshot', () => {
  const { asFragment } = render(
    <HighContrastStore value={defaultHighContrastState}>
      <Pronunciation />
    </HighContrastStore>
  );

  expect(asFragment()).toMatchSnapshot();
})

test('Matches snapshot with high contrast', () => {
  const { asFragment } = render(
    <HighContrastStore value={{
      state: true
    }}>
      <Pronunciation />
    </HighContrastStore>
  );

  expect(asFragment()).toMatchSnapshot();
})

test('Clicking plays audio', () => {
  window.HTMLMediaElement.prototype.play = jest.fn();

  const { getByLabelText } = render(
    <HighContrastStore value={{
      state: true
    }}>
      <Pronunciation />
    </HighContrastStore>
  );

  const button = getByLabelText('Play sound');

  fireEvent.click(button);

  expect(window.HTMLMediaElement.prototype.play).toBeCalled();
})
