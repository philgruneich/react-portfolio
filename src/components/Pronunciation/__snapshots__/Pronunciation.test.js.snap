// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Play sound"
    class="Pronunciationstyles__PronunciationButton-ehdec3-0 cDuaSV"
    type="button"
  >
    <audio
      crossorigin="anonymous"
      type="audio/mp3"
    />
    <svg
      class="Pronunciationstyles__PronunciationSvg-ehdec3-1 gAtKcF"
      viewBox="0 0 49 49"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Pronunciationstyles__PronunciationPath-ehdec3-2 bWoeOm"
        d="M34.112 38.514l-15.425-2.283-6.146 3.548-7.617-13.194 6.138-3.543L16.72 8.389l17.392 30.125zM13.48 25.108l-4.457 2.574 4.616 7.996 4.45-2.57 10.32 1.53-11.153-19.316c-1.259 3.263-2.516 6.526-3.776 9.786zM28.98 8.368c7.962.133 14.724 8.632 12.477 16.63-.895 3.189-3.026 5.975-5.91 7.715l-1.432.839-1.516-2.59c5.236-3.066 8.062-9.689 4.794-14.883-2.71-4.306-8.834-5.985-13.47-3.424l-1.427.811-1.483-2.607 1.442-.82c2.022-1.117 3.93-1.682 6.524-1.67zm-.055 5.795c6.032.1 10.28 9.572 3.66 13.568l-1.369.8-1.517-2.588c.922-.538 1.906-1.017 2.549-1.848 2.288-2.953-1.401-8.643-5.493-6.382l-1.361.773-1.483-2.608c1.576-.896 3.216-1.722 5.014-1.715z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Matches snapshot with high contrast 1`] = `
<DocumentFragment>
  <button
    aria-label="Play sound"
    class="Pronunciationstyles__PronunciationButton-ehdec3-0 cDuaSV"
    type="button"
  >
    <audio
      crossorigin="anonymous"
      type="audio/mp3"
    />
    <svg
      class="Pronunciationstyles__PronunciationSvg-ehdec3-1 gAtKcF"
      viewBox="0 0 49 49"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Pronunciationstyles__PronunciationPath-ehdec3-2 etNdUX"
        d="M34.112 38.514l-15.425-2.283-6.146 3.548-7.617-13.194 6.138-3.543L16.72 8.389l17.392 30.125zM13.48 25.108l-4.457 2.574 4.616 7.996 4.45-2.57 10.32 1.53-11.153-19.316c-1.259 3.263-2.516 6.526-3.776 9.786zM28.98 8.368c7.962.133 14.724 8.632 12.477 16.63-.895 3.189-3.026 5.975-5.91 7.715l-1.432.839-1.516-2.59c5.236-3.066 8.062-9.689 4.794-14.883-2.71-4.306-8.834-5.985-13.47-3.424l-1.427.811-1.483-2.607 1.442-.82c2.022-1.117 3.93-1.682 6.524-1.67zm-.055 5.795c6.032.1 10.28 9.572 3.66 13.568l-1.369.8-1.517-2.588c.922-.538 1.906-1.017 2.549-1.848 2.288-2.953-1.401-8.643-5.493-6.382l-1.361.773-1.483-2.608c1.576-.896 3.216-1.722 5.014-1.715z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
